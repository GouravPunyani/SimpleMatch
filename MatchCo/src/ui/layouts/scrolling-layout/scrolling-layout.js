// @flow

import React, { Component } from 'react'
import { ScrollView, StyleSheet } from 'react-native'
import { color, spacing } from '../../theme'

export type ScrollingLayoutProps = {
  children?: React.Children,
  containerContentStyle?: any
};

export class ScrollingLayout
  extends Component<void, ScrollingLayoutProps, void> {
  render () {
    const style = [styles.wrapper]
    const containerContentStyle = [styles.containerContent]
    if (this.props.style) {
      style.push(this.props.style)
    }
    if (this.props.containerContentStyle) {
      containerContentStyle.push(this.props.containerContentStyle)
    }
    return (
      <ScrollView
        style={style}
        contentContainerStyle={containerContentStyle}
        keyboardShouldPersistTaps='handled'
      >
        {this.props.children}
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    backgroundColor: color.background
  },
  containerContent: {
    padding: spacing.medium
  }
})
