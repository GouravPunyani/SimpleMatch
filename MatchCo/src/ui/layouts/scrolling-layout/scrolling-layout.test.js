// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { ScrollingLayout as Comp } from './scrolling-layout'

test('renders correctly', () => {
  const tree = renderer.create(<Comp />)
  expect(tree).toMatchSnapshot()
})
