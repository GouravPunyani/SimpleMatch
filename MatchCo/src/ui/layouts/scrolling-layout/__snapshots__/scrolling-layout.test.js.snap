// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "padding": 8,
    }
  }
>
  <View />
</RCTScrollView>
`;
